spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/support/AppResourceCommon.java

```diff
@@ -34,10 +35,66 @@
 import org.springframework.util.StringUtils;
 
 /**
- * @author Mark Pollack
- * @author Ilayaperumal Gopinathan
+ * @author Christian Tzolov
  */
-public class ResourceUtils {
+public class AppResourceCommon {
+
+	/**
+	 * the maven properties to use in case of maven resource
+	 */
+	private MavenProperties mavenProperties;
+
+	/**
+	 *
+	 */
+	private ResourceLoader resourceLoader;
+
+
+	public AppResourceCommon(MavenProperties mavenProperties, ResourceLoader resourceLoader) {
```

Comment:
Could you add null asserts for props and loader?

Reply:
I will for the props, but the (metadata)resourceLoader can actually be null for Docker apps